### DISCUSSION TITLE
Error install tensorflow di R dengan chipset m1/m2

### DISCUSSION THREAD

**Discussion by trisaputrau** [teamalgoritma/community#622] [Q&A] [answered]
mengikuti saran di https://stackoverflow.com/questions/71355902/how-to-install-keras-using-r-studio-on-a-mac-os-computer 

setelah menggunakan `which -a python python3` dan menggunakan pathnya masih tetap error
<img width="1173" alt="Screenshot 2023-04-11 at 17 29 14" src="https://user-images.githubusercontent.com/119938911/231134456-f6e8750f-415b-42de-9ff1-cec0412480e0.png">

**Comment by hwulanayu** [comment, marked as answer]
Sepertinya itu hanya warning saja pak. Artinya nanti Rstudio akan menggunakan path python yang Bapak pilih di use_python. Beberapa langkah selanjutnya yg bisa dicoba:
1. Buka Terminal, aktifkan env tensorflow_m2, ketik command 'which python'. output dari command akan bapak gunakan nanti di Rstudio, jadi biarkan saja dulu terminalnya 
2. Buka RStudio, Install library keras dan tensorflow di Rstudio. Pake yang di panel kanan bawah aja pak, di bagian tab Packages > Install > ketik librarynya
3. Restart Session dan silakan Bapak coba kode di bawah ini **satu per satu**:
```
reticulate::use_python("isi dengan path yg dari output terminal which python")
reticulate::use_condaenv("tensorflow_m2", required = TRUE)

library(tensorflow)
tf_config() 
tf_version() 
tf$config$list_logical_devices()
```
4. Jika tidak ada error dari semua line code di atas, silakan buka Tools > Global Options > Python. Di Bagian Python Interpreter, hapus path default, ganti dengan output path python dari tf_config() di atas. Contohnya, kalo outputnya seperti ini,
```
> tf_config() 
TensorFlow v2.9.0 (/opt/miniconda3/envs/tensorflow_m1/lib/python3.8/site-packages/tensorflow)
Python v3.8 (/opt/miniconda3/envs/tensorflow_m1/bin/python3.8)
```
berarti yang Bapak masukan di Global Options adalah yg "/opt/miniconda3/envs/tensorflow_m1/bin/python3.8". (minus tanda petiknya).
5. Kalau sudah, coba Close Rstudionya dan buka Rproject materi NN yang menggunakan tensorflownya. Run All dan pastikan semua code bisa bekerja.
6. Jika masih ada error silakan screenshot kembali errornya di discussion ini

**Comment by trisaputrau** [comment]
Masih error ketika 

`> tf_config() 
Error in system2(command = python, args = shQuote(script), stdout = TRUE,  : 
  error in running command`

Tapi udah diatasi, ternyata butuh di set environmentnya dulu di R pakai

`Sys.setenv(RETICULATE_PYTHON = "/Users/trisaputraungko/miniconda3/envs/tensorflow_m2/bin/python")`

hasilnya:

`> tf_config() 
TensorFlow v2.9.0 (~/miniconda3/envs/tensorflow_m2/lib/python3.8/site-packages/tensorflow)
Python v3.8 (~/miniconda3/envs/tensorflow_m2/bin/python)`

<img width="1174" alt="Screenshot 2023-04-11 at 18 11 17" src="https://user-images.githubusercontent.com/119938911/231143166-c5ab6d7a-aa20-4ca5-a047-a799210c10da.png">
